tests/application/cms/test_views.py

```diff
@@ -995,3 +949,139 @@ class TestChartBuilder(_TestVisualisationBuilder):
 
 class TestTableBuilder(_TestVisualisationBuilder):
     ROUTE_NAME = "cms.create_table"
+
+
+class TestRemoveDataSourceView:
+    def setup(self):
+        self.saved_config = {**current_app.config}
+
+    def teardown(self):
+        current_app.config = {**self.saved_config}
+
+    def test_login_required(self, test_app_client):
+        measure_version = MeasureVersionFactory()
+
+        res = test_app_client.post(
+            url_for(
+                "cms.remove_data_source",
+                topic_slug=measure_version.measure.subtopic.topic.slug,
+                subtopic_slug=measure_version.measure.subtopic.slug,
+                measure_slug=measure_version.measure.slug,
+                version=measure_version.version,
+                data_source_id=measure_version.data_sources[0].id,
+            ),
+            follow_redirects=False,
+        )
+
+        assert res.status_code == 302
+        assert parse.urlparse(res.location).path == url_for("security.login")
+
+    def test_department_user_needs_specific_access_to_measure(self, test_app_client, logged_in_dept_user):
+        measure_version = MeasureVersionFactory()
+
+        res = test_app_client.post(
+            url_for(
+                "cms.remove_data_source",
+                topic_slug=measure_version.measure.subtopic.topic.slug,
+                subtopic_slug=measure_version.measure.subtopic.slug,
+                measure_slug=measure_version.measure.slug,
+                version=measure_version.version,
+                data_source_id=measure_version.data_sources[0].id,
+            ),
+            follow_redirects=True,
+        )
+
+        assert res.status_code == 403
```

Comment:
Should we assert where the redirection goes?

Reply:
This test doesn't result in a redirect - it's just an unauthorised response.

Reply:
Should we set `follow_redirects=False` then?